p2p/client.go

```diff
@@ -394,17 +503,43 @@ func (c *Client) NewTxValidator() GossipValidator {
 	}
 }
 
-func (c *Client) gossipedBlockReceived(msg *GossipMessage) {
-	var gossipedBlock GossipedBlock
-	if err := gossipedBlock.UnmarshalBinary(msg.Data); err != nil {
+// blockSyncReceived is called on reception of new block via block-sync protocol
+func (c *Client) blockSyncReceived(block *P2PBlockEvent) {
+	err := c.localPubsubServer.PublishWithEvents(context.Background(), *block, map[string][]string{EventTypeKey: {EventNewBlockSyncBlock}})
+	if err != nil {
+		c.logger.Error("Publishing event.", "err", err)
+	}
+	// Received block is cached and  no longer needed to request using block-sync
+	if c.conf.BlockSyncEnabled {
```

Comment:
assuming `BlockSyncEnabled=false`, and we never start the blockSync service, is this method still gonna be called?

Reply:
true. removed